Nulecule.yaml

```diff
@@ -0,0 +1,95 @@
+# Put all metadata here (no more separate id, specversion objects, etc.
+metadata:
+  name: Hello World
+  id: helloworld
+  appversion: 0.0.1
+  spec: 1.0.0
+  description: Application
+
+# All provider details / what provider to orchestrate to. Because let's face it, 99% of the time the user is only focused on one specified provider when deploying Atomic App
+provider:
```

Comment:
I don't like the part that Nulecule related keys: ``metadata``, ``provider`` share the same level as the components: ``db``, ``etherpad``, etc. This might lead to conflicts and confusion. We need to group one of them.